### PR TITLE
BUG: caching broken due to changed callback.

### PR DESCRIPTION
**Pull request by rainwoodman**
This PR implements

https://github.com/dask/dask/pull/4460#issuecomment-469374479

It's an alternative take to #4460.

- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @crepererum @jcrist for review

**Comment by jcrist** [comment]
LGTM, provided tests pass.

File: dask/local.py [hunk 1]
```diff
@@ -404,17 +404,18 @@ def get_async(apply_async, num_workers, dsk, result, cache=None,
         _, _, pretask_cbs, posttask_cbs, _ = unpack_callbacks(callbacks)
         started_cbs = []
         succeeded = False
+        state = {} # if start_state_from_dask fails, we will have something
+                   # to pass to the final block.
```
Comment (jakirkham) [inline dask/local.py]: Flake8 seems to dislike the indented comment here. Maybe move the comments above the `state` line?

**Comment by crepererum** [comment]
@rainwoodman could you please add a regression test? Apart from that, thanks, good catch! 👍

**Comment by rainwoodman** [comment]
A new regression test is a bit hard. I couldn't find a minimal reliable reproducer. 

It only fails when I run our entire test suite. It does not fail if I run the failed test just by itself.

On the other hand your previous regression test is still included.

**Pull request merged**